### PR TITLE
Fix Playlist.iter_fans, improve tests to catch similar errors

### LINKED ISSUE #135: Playlist.iter_fans iterates over tracks, not fans
Discovered when adding `Podcast` and `Episode` functionality for #132 (in #134)

`Playlist.iter_fans` actually iterates over `track`s, not `fan`s

https://github.com/browniebroke/deezer-python/blob/99f01b79d91811d69e3b95a89ea8c38c84f2b404/deezer/resources.py#L326-L332

I'll introduce a PR to fix this. The reason for creating separate issues/PRs for all of these issues is just so it doesn't all get mixed into the same thing (when they're all quite different problems).

Testing obviously didn't catch this, so in the PR to fix this I'll also beef up the generator testing (all the generator tests currently assert is that the type is actually `GeneratorType`). From something like this...
https://github.com/browniebroke/deezer-python/blob/99f01b79d91811d69e3b95a89ea8c38c84f2b404/tests/test_resources.py#L274
...to something like this:
```python3
# test generator
fans_generator = playlist.iter_fans()
assert type(fans_generator) == GeneratorType
fan = next(fans_generator)
assert fan.name == "laurentky"
count = 1
while 1:
    assert isinstance(fan, deezer.resources.User)
    try:
        fan = next(fans_generator)
        count += 1
    except StopIteration:
        break
assert count == 3
```
Please let me know if you know of a better way to test these, as (unless I think of any better ways) the code above is probably what I'll be using - I'll probably make a helper function for most of that. My aim with coding tests for the generators is to make them equal to the tests for lists - if that was the case already, the `Playlist.iter_fans` typo would've been caught.

### PR DESCRIPTION
**Pull request by hithomasmorelli**
Fixes #135 

`Playlist.iter_fans` now iterates on fans (and not tracks, as it has been doing).
Tests for generators (`iter_*`) now perform the same level of tests as with the lists (`get_*`). The updated tests will prevent this error from reoccurring.

- [X] Include tests for bug fix and new functionality
- [X] Updated documentation for new feature

### CODE REVIEW

File: tests/test_resources.py [hunk 1]
```diff
@@ -213,68 +324,143 @@ def test_user_albums(self, client):
...
         assert len(tracks) == 4
         for track in tracks:
             assert isinstance(track, deezer.resources.Track)
         assert tracks[0].title == "Skanky Panky"
-        assert type(playlist.iter_tracks()) == GeneratorType
+
+        # tests generator
+        tracks_generator = playlist.iter_tracks()
+        assert type(tracks_generator) == GeneratorType
+        track = next(tracks_generator)
```
Comment (browniebroke) [inline tests/test_resources.py:421]: Maybe the rest of this test would be more readable if we were casting it to a list before doing assertion on the content:

```
playlist_tracks = playlist.iter_tracks()
assert type(playlist_tracks) == GeneratorType
playlist_tracks = list(playlist_tracks)
assert len(playlist_tracks) == 5
assert all(isinstance(track, deezer.resources.Track) for track in playlist_tracks)
```

File: tests/test_resources.py [hunk 2]
```diff
@@ -213,68 +324,143 @@ def test_user_albums(self, client):
...
+            assert isinstance(track, deezer.resources.Track)
+            try:
+                track = next(tracks_generator)
+                count += 1
+            except StopIteration:
+                break
+        # this is weird (the count being different), it seems there's a sort of
+        # partially-hidden track that only shows up when using the method in
+        # iter_tracks
+        assert count == 5
```
Comment (browniebroke) [inline tests/test_resources.py:434]: That's weird, I looked at the content of the response and it was:

```
[<Track: Skanky Panky>,
 <Track: More Dance Music>,
 <Track: Basin Street Blues>,
 <Track: Celebrity Death Match (feat. Kid Koala)>,
 <Track: Celebrity Death Match (feat. Kid Koala)>]
```

There is potentially a bug somewhere, but it's strange that it doesn't affect others `iter_...` methods, as the Playlist's implementation hasn't got much specific.

**Review by browniebroke — COMMENTED** [review]
Looks good to me otherwise, good work again 💪

File: tests/test_resources.py (hunk 2 above)
Comment (hithomasmorelli) [reply tests/test_resources.py:434]: This is a bug with the Deezer API itself (I think). The final result is gained by `iter_tracks` performing a second call to the endpoint, this time with `index=4`. I've no idea why it happens, given that the correct index (for the number of previous results) is being passed to the API.

I'm afraid I probably won't be much use in solving this one!

File: tests/test_resources.py (hunk 1 above)
Comment (hithomasmorelli) [reply tests/test_resources.py:421]: LGTM - I'll edit and commit it

File: tests/test_resources.py (hunk 2 above)
Comment (browniebroke) [reply tests/test_resources.py]: Yea, no need to worry too much about this one, looks out of scope anyway.

**Review by browniebroke — APPROVED** [review]

**Pull request merged**